### PR TITLE
bugfix: check_format script, tokenInLine helper function

### PR DESCRIPTION
**Pull request by stedsome**
Signed-off-by: Yifan Yang <needyyang@google.com>
This looks like a bug to me. @sunjayBhatia do you mind confirming this? 
The effect of this is that this function incorrectly returns false when the token in question is the first one in a line. 

This was found during my attempt here https://github.com/envoyproxy/envoy/pull/12421. 
Commit Message:
Additional Description:
Risk Level: low, should correct an incorrect behavior. 
Testing: All existing tests have passed 
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
@stedsome can you add a test to the code format checker that demonstrates this issue?

**Comment by stedsome** [comment]
@sunjayBhatia 
If you checkout the current master, where new restrictions on std::any/optional/variant are enforced, the following testcase will not be able to catch the issue. 

```
#include <any>

namespace Envoy {
std::any foo;
} // namespace Envoy
```

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -567,7 +567,7 @@ def tokenInLine(token, line):
   index = 0
   while True:
     index = line.find(token, index)
-    if index < 1:
+    if index < 0:
```
Comment (dio) [inline tools/code_format/check_format.py:570]: Mind to put your comment in the commit description here too? Plus if you can add a case example here will be good. Thanks!

Comment (stedsome) [reply tools/code_format/check_format.py:570]: @dio Just to clarify, do you mean putting the example case in the comment section?

Comment (dio) [reply tools/code_format/check_format.py:570]: Yes, sorry for not very clear earlier. Put an inline comment here, why we need to branch when the index is 1 (and not 0), and what is the implication as you have described in the PR description above.

Comment (stedsome) [reply tools/code_format/check_format.py:574]: Comments are added. Let me know if this is okay now.

**Pull request merged**